### DISCUSSION TITLE
Relay plugin do not expose node/nodes queries

### DISCUSSION THREAD

**Discussion by TLadd** [hayes/pothos#749] [Q&A] [answered]
I'm attempting to use the pothos Relay plugin in conjunction with the Apollo Federation plugin. One of the tricky things about Apollo Federation and Relay together is implementing the `node` query to load any Node type by id. As best I can tell, the current best workaround is to create a separate subgraph with the `node` query that also declares all `Node` types with just the `id` field. See https://github.com/apollographql/federation/issues/1067 for further discussion. When processing a `node` query, the router sends the query first to the separate node resolution subgraph, and then resolves the additional fields from the subgraphs that define them.

With the pothos relay plugin, when I define a type using `node` or `prismaNode`, it automatically adds the `node` and `nodes` query to the subgraph schemas. So I end up with the `node` query declared in both the node resolution subgraph as well as the subgraph that is declaring the fields for the type. This situation would result in the router being able to send the query to either subgraph, one of which wouldn't actually know about all types (in the case of multiple subgraphs contributing Node types).

Ideally, I could prevent the `node` and `nodes` queries from being output to the subgraph schemas. Would something like that be on the table? I think the other way for me to go would be to stop using `prismaNode` in the subgraphs that declares the fields altogether and manually implement the Node interface.

**Comment by hayes** [comment, marked as answer]
Just opened a PR that adds an option for omitting those fields

**Reply by hayes** [reply]
https://github.com/hayes/pothos/pull/751
